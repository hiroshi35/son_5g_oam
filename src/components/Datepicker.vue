<template>
  <div>
    <div v-if="icon" class='input-group date' ref="element">
      <input type='text' class="form-control" :placeholder="placeholder" v-model="value">
      <span class="input-group-addon" v-html="customIcon"></span>
    </div>
    <input v-else type="text" class="form-control" :placeholder="placeholder" ref="element" v-model="value">
  </div>

</template>
<script>
  import 'moment'
  import 'eonasdan-bootstrap-datetimepicker'

  export default {
    props: {
      placeholder: {
        type: String,
        default: 'Enter Date'
      },
      options: {
        type: Object
      },
      icon: {
        type: Boolean,
        default: false
      },
      customIcon: {
        type: String,
        default: '<span class="glyphicon glyphicon-calendar"></span>'
      },
      format: {
        type: String,
        default: 'DD-MM-YYYY HH:mm'
      },
      value: {
        type: String,
        default: ''
      }
    },
    computed: {
      optionObj () {
        if (this.options) {
          return this.options
        }
        let obj = {}
        if (this.format) {
          obj.format = this.format
        }
        return obj
      }
    },
    mounted () {
      let vm = this
      $(this.$refs.element).datetimepicker(this.optionObj).on('dp.change', function (e) {
        console.log('sd');
        vm.$emit('input', e.date.format(vm.format))
      })
    }
  }
</script>

<style lang="scss">
  @import "~eonasdan-bootstrap-datetimepicker/build/css/bootstrap-datetimepicker";
</style>
