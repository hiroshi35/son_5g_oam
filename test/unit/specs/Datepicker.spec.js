import { shallow } from 'vue-test-utils'
import Datepicker from '@/components/Datepicker.vue'
import {datetimepicker, datetimepickerOn} from 'jquery'

describe('Datepicker.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Datepicker)
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders .input-group-addon', () => {
    wrapper.setProps({icon: true})
    expect(wrapper.find('.input-group-addon').is('span')).toMatchSnapshot()
  })
  test('$().datetimepicker() should be called', () => {
    expect(datetimepicker).toBeCalled()
  })
  test('$().datetimepicker().on() should be called with correct argument', () => {
    expect(datetimepickerOn).toBeCalled()
    expect(datetimepickerOn.mock.calls[0][0]).toBe('dp.change')
  })
})
