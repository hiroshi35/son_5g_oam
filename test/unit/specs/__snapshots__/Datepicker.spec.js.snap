// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker.vue has the expected html structure 1`] = `
<div>
  <input
    class="form-control"
    placeholder="Enter Date"
    type="text"
  />
</div>
`;

exports[`Datepicker.vue renders .input-group-addon 1`] = `true`;
